controllers/chat/ChatController.js

```diff
@@ -181,12 +181,13 @@ module.exports = {
       });
       return res.status(200).json(responseSuccess('Success retrieve channels', channels));
     } catch (error) {
-      await client.disconnectUser();
       return res.status(error.statusCode ?? error.status ?? 400).json({
         status: 'error',
         code: error.statusCode ?? error.status ?? 400,
         message: error.message
       });
+    } finally {
+      await client.disconnectUser();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Good job adding a `finally` block to ensure that `client.disconnectUser()` is always called, even if an error occurs. This is a good practice as it helps prevent potential memory leaks and ensures that resources are properly cleaned up.


<!-- This is an auto-generated comment by OSS CodeRabbit -->